### DISCUSSION TITLE
How to get docs sidebar items

### DISCUSSION THREAD

**Discussion by jagoral** [facebook/docusaurus#7457] [Q&A]
Hi, I checked the documentation and the source code, and I couldn't find a way to get all the docs sidebar items. I need to have the list of [PropSidebarItem](https://github.com/facebook/docusaurus/blob/20f333bb504b1f59693f5c65bffa0fafab2a44da/packages/docusaurus-plugin-content-docs/src/sidebars/types.ts#L189), because basing on them I compute a svg image with the overview of all docs pages. Does anyone have idea how to accomplish it? The only thing which comes to my mind is to write a custom plugin. In `usePluginData('docusaurus-plugin-content-docs')` I almost had the needed data (`versions[0].docs`), but the title of each sidebar item was missing

**Comment by jagoral** [comment]
Okay, I found it. I had to just `import { useDocsSidebar } from "@docusaurus/theme-common";` within docs context :).

**Reply by slorber** [reply]
Yes 👍 keep in mind it's still an internal API that may change or be renamed.

(although it's unlikely we'll remove this feature, just please avoid building a custom community plugin on top of it)

**Reply by roberthusak** [reply]
Btw. it's `import {useDocsSidebar} from "@docusaurus/theme-common/internal";` in the version 2.0.1.

**Reply by usefulthink** [reply]
Quick update for anyone stumbling across this: as of version 3.5.0 this is no longer found in theme-common, but in plugin-content-docs (although I have no idea if that is still considered an internal API):

```js
import {useDocsSidebar} from '@docusaurus/plugin-content-docs/client';
```
